api/v3/utils.php

```diff
@@ -1931,17 +1998,23 @@ function _civicrm_api3_api_match_pseudoconstant(&$params, $entity, $fieldName, $
     }
     $options = civicrm_api($entity, 'getoptions', array('version' => 3, 'field' => $fieldInfo['name'], 'context' => 'validate'));
     $options = CRM_Utils_Array::value('values', $options, array());
+
+    if (count($options) == 0 && isset($pseudoconstant['table'])) {
+        $pseudoParams = $pseudoconstant;
+        unset($pseudoParams['table']);
+        $options = CRM_Core_PseudoConstant::get(CRM_Core_DAO_AllCoreTables::getClassForTable($pseudoconstant['table']), $fieldName, $pseudoParams);
```

Comment:
I want to clarify that api.getoptions *does* understand and should correctly handle pseudoconstants that specify "table" and *does not* require the corresponding BAO to have a "buildOptions" method (since all BAOs inherit this method from CRM_Core_DAO it is optional but not necessary for them to override it).

So I still don't understand the problem you are trying to solve, but I suspect this is not the correct solution. If api.getoptions is messing up we should fix it rather than working around it.